### PR TITLE
For consistency make repo search sort options case sensitive

### PR DESCRIPTION
**Pull request by 6543**
as title.

smal int missed by #30876

### CODE REVIEW

**Review by delvh — APPROVED** [review]
And here I was thinking you made the **result** case-insensitive, not only the parameter…

File: routers/api/v1/repo/repo.go [hunk 1]
```diff
@@ -178,7 +178,8 @@ func Search(ctx *context.APIContext) {
 		opts.IsPrivate = optional.Some(ctx.FormBool("is_private"))
 	}
 
-	sortMode := ctx.FormString("sort")
+	// as sort is an url param, we should be case insensitive
+	sortMode := strings.ToLower(ctx.FormString("sort"))
```
Comment (wxiaoguang) [inline routers/api/v1/repo/repo.go]:
```suggestion
	sortMode := ctx.FormString("sort")
```
Meanless

**Review by wxiaoguang — CHANGES_REQUESTED** [review]

File: routers/api/v1/repo/repo.go (hunk 1 above)
Comment (6543) [reply routers/api/v1/repo/repo.go]: please tell me why it is

Comment (wxiaoguang) [reply routers/api/v1/repo/repo.go]: Why "as sort is an url param, it should be case insensitive"? What's the real use-case? Any other servers do so?

Why you (and the server) would like to pay the cost for end-developer's incorrect-case mistake?

Comment (6543) [reply routers/api/v1/repo/repo.go]: that's a cursed question ...
... in this case you could also argue why we do trim issue titles on creation ... it should be the users vault if the title contains spaces ...

Comment (wxiaoguang) [reply routers/api/v1/repo/repo.go]: > ... in this case you could also argue why we do trim issue titles on creation

That's different. Title is inputted by end users (UGC, user-generated-content). But here, these "sort modes" are "constant values".

If you considered the "sort mode" should be case-insensitive, then:

* you are inventing a new mechanism which never appeared in Gitea code base.
* you only hacky-patched this parameter in this function, but all (almost) other functions do not use case-insensitive constant values. And even in this function: `FormBool` doesn't accept `tRUE`, `sortOrder` doesn't use case-insensitive, the system is just breaking up to different mechanisms.

Comment (6543) [reply routers/api/v1/repo/repo.go]: well for the sake of consistence ...

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**